### PR TITLE
Update Microsoft.Build dependency for Dev17.5

### PR DESCRIPTION
**Pull request by KevinRansom**
Microsoft.Build has a dependency on an invalid version of System.Cryptography.Xml.  This PR updates our dependency to MSBuild 17.5.0 which references a correct version of System.Cryptography.Xml

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -13,8 +13,12 @@
     <add key="dotnet7" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet7/nuget/v3/index.json" />
     <add key="dotnet7-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet7-transport/nuget/v3/index.json" />
     <add key="dotnet-public" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public/nuget/v3/index.json" />
+    <add key="nuget-build" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/nuget-build/nuget/v3/index.json" />
     <add key="vssdk" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk/nuget/v3/index.json" />
+    <add key="vssdk-archived" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk-archived/nuget/v3/index.json" />
```
Comment (psfinaki) [inline NuGet.config:18]: Wait are we sure we want to add some archived feeds here?

Comment (vzarytovskii) [reply NuGet.config:18]: Yes, it's the case in all newer release branches

**Review by T-Gro — APPROVED** [review]

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**